docs/source/user_guide/spectrum_merging.rst

```diff
@@ -0,0 +1,341 @@
+Spectra Merge Algorithm
+*************************
+
+OpenMS provides spectra merging and averaging algorithms in :py:class:`~.SpectraMerger` class. Spectra merging is to merge multiple related spectra into a single one - thus, often we end up with a reduced number of spectra.
+For instance, MS1 spectra within a pre-defined retention time window or MS2 spectra from the same precursor ion. On the other hand, spectra averaging averages neighbouring spectra for each spectrum.
+Thus, the number of spectra remains the same after spectra averaging. Both merging and averaging attempt to increase the quality of spectrum by increasing its signal to noise ratio. 
+
+Spectra merging and averaging are implemented in SpectraMerger in pyOpenMS, which provides two merging (block wise and precursor method - see below) and two averaging methods (gaussian and tophat - see below).
+For merging, we can use 
+
+- mergeSpectraBlockWise
+- mergeSpectraPrecursors
+
+and for averaging, we use 
+
+- average 
+
+with two different options: gaussian and tophat methods. :py:class:`~.SpectraMerger` inherits :py:class:`~.DefaultParamHandler`. Thus you could list the full set of parameters as described in `Parameter handling <parameter_handling.html>`_.
+
+
+Loading the Raw Data
+--------------------
+
+Let's take a look at the different algorithms in the examples below. First we download a test MS/MS dataset.
+
+.. code-block:: python
+  :linenos:
+
+  # retrieve data
+  from urllib.request import urlretrieve
+  import pyopenms as oms
+  import matplotlib.pyplot as plt
+
+  gh = "https://raw.githubusercontent.com/OpenMS/pyopenms-docs/master"
+  urlretrieve(gh + "/src/data/small.mzML", "test.mzML")
+
+  # load MS data and store as MSExperiment object
+  exp = oms.MSExperiment()
+  oms.MzMLFile().load('test.mzML', exp)
+
+
+
+Block wise spectra merging
+-----------------------
+Our first example merges MS1 spectra block wise.
+
+.. code-block:: python
+  :linenos:
+
+  spectra = exp.getSpectra()
+
+  spectra_ms1 = [s for s in spectra if s.getMSLevel() == 1]
+  print(f'Number of MS1 spectra before merge are {len(spectra_ms1)}')
+
+  # merges blocks of MS1
+  merger = oms.SpectraMerger()
+  merger.mergeSpectraBlockWise(exp)
+
+  spectraMerged = exp.getSpectra()
+  spectraMerged_ms1 = [s for s in spectraMerged if s.getMSLevel() == 1]
+  print(f'Number of MS1 spectra after merge are {len(spectraMerged_ms1)}')
+  # store merged spectra in the disk
+  oms.MzMLFile().store("blockwiseMerged.mzML", exp)
+
+  fig, axs = plt.subplots(6)
```

Comment:
maybe a comment about why there are six subplots: Setting up subplots for five original spectra and merged spectrum